chore: Update users.yml with current Screens eng team

robbie-sundstrom opened the pull request:
Update the users.yml files to be up to date with changes to the Screens team
- Remove Paul and Christian since they've left TID 
- Add myself and Dean

krisrjohnson21 reviewed (commented):
@robbie-sundstrom this looks great! We do need to take one more step and ensure that all deleted users are absent from the config. 

Could you follow the structure of [this file in devops](https://github.com/mbta/devops/blob/main/ansible/roles/gateway/vars/main.yml) to add a `deleted_users` group and ensure that the `username` values for all deleted users (Christian and Paul) are included?

robbie-sundstrom commented:
@krisrjohnson21 I added it to the individual files where I removed these users, but I also see that there is a `users_deleted` list in `linux/group_vars/all/users.yml`. Wondering if it makes sense to consolidate there.
